flair/data.py

```diff
@@ -433,61 +433,71 @@ def __len__(self):
-    """
+class EntityLinkingCandidate:
+    """Represent a single candidate returned by a CandidateGenerator"""
 
     def __init__(
         self,
-        data_point: DataPoint,
         concept_id: str,
         concept_name: str,
+        database_name: str,
         score: float = 1.0,
         additional_ids: Optional[Union[List[str], str]] = None,
-        database: Optional[str] = None,
     ):
         """
-        Initializes the label instance.
-
-        :param data_point: Data point / span the label refers to
         :param concept_id: Identifier of the entity / concept from the knowledge base / ontology
         :param concept_name: (Canonical) name of the entity / concept from the knowledge base / ontology
         :param score: Matching score of the entity / concept according to the entity mention
         :param additional_ids: List of additional identifiers for the concept / entity in the KB / ontology
-        :param database: Name of the knowlege base / ontology
+        :param database_name: Name of the knowlege base / ontology
         """
-        super().__init__(data_point, concept_id, score)
+        self.concept_id = concept_id
         self.concept_name = concept_name
-        self.database = database
-
-        if isinstance(additional_ids, str):
-            additional_ids = [additional_ids]
+        self.database_name = database_name
+        self.score = score
         self.additional_ids = additional_ids
 
-    def spawn(self, value: str, score: float = 1.0):
-        return EntityLinkingLabel(
-            data_point=self.data_point,
-            concept_id=value,
-            score=score,
-            concept_name=self.concept_name,
-            additional_ids=self.additional_ids,
-            database=self.database,
-        )
+
+class EntityLinkingLabel(Label):
+    """
+    Label class models entity linking annotations. Each entity linking label has a data point it refers
+    to as well as the identifier and name of the concept / entity from a knowledge base or ontology.
+    Optionally, additional concepts identifier and the database name can be provided.
+    """
+
+    def __init__(self, data_point: DataPoint, candidates: List[EntityLinkingCandidate]):
+        """
+        Initializes the label instance.
+        :param data_point: Data point / span the label refers to
+        :param candidates: **sorted** list of candidates from candidate generator
+        """
+
+        def is_sorted(lst, key=lambda x: x, comparison=lambda x, y: x > y):
+            for i, el in enumerate(lst[1:]):
+                if comparison(key(el), key(lst[i])):
+                    return False
+            return True
+
+        # candidates must be sorted, regardless if higher is better or not
+        assert is_sorted(candidates, key=lambda x: x.score) or is_sorted(
+            candidates, key=lambda x: x.score, comparison=lambda x, y: x < y
```

Comment:
In theory two concepts can get the same score. Therefore we should lessen the constraint to less-/greater-equal here.